modules/azure/client_factory.go

```diff
@@ -967,3 +1002,53 @@ func getBaseURI() (string, error) {
 	}
 	return baseURI, nil
 }
+
+// getArmAppContainersClientFactory gets an arm app containers client factory
+func getArmAppContainersClientFactory(subscriptionID string) (*armappcontainers.ClientFactory, error) {
+	targetSubscriptionID, err := getTargetAzureSubscription(subscriptionID)
+	clientCloudConfig, err := getClientCloudConfig()
+	if err != nil {
+		return nil, err
+	}
+	cred, err := azidentity.NewDefaultAzureCredential(&azidentity.DefaultAzureCredentialOptions{
+		ClientOptions: azcore.ClientOptions{
+			Cloud: clientCloudConfig,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return armappcontainers.NewClientFactory(targetSubscriptionID, cred, &arm.ClientOptions{
+		ClientOptions: policy.ClientOptions{
+			Cloud: clientCloudConfig,
+		},
+	})
+}
+
+func getClientCloudConfig() (cloud.Configuration, error) {
+	envName := getDefaultEnvironmentName()
+	switch strings.ToUpper(envName) {
+	case "AZURECHINACLOUD":
+		return cloud.AzureChina, nil
+	case "AZUREUSGOVERNMENTCLOUD":
+		return cloud.AzureGovernment, nil
+	case "AZUREPUBLICCLOUD":
+		return cloud.AzurePublic, nil
+	case "AZURESTACKCLOUD":
+		env, err := autorestAzure.EnvironmentFromName(envName)
+		if err != nil {
+			return cloud.Configuration{}, err
+		}
+		c := cloud.Configuration{
+			ActiveDirectoryAuthorityHost: env.ActiveDirectoryEndpoint,
+			Services: map[cloud.ServiceName]cloud.ServiceConfiguration{
+				cloud.ResourceManager: {
+					Audience: env.TokenAudience,
+					Endpoint: env.ResourceManagerEndpoint,
+				},
+			},
+		}
+		return c, err
```

Comment:
nit: return c, nil 

clear that there's not error.

Reply:
changed to `return c, nil`